subprojects/core/src/integTest/groovy/org/gradle/api/tasks/TaskRemovalIntegrationTest.groovy

```diff
@@ -101,8 +104,30 @@ class TaskRemovalIntegrationTest extends AbstractIntegrationSpec {
         """
 
         then:
+        executer.expectDeprecationWarning()
         fails ":bar"
         failure.assertThatCause(Matchers.startsWith("Tried to remove model 'tasks.foo' but it is depended on by: 'tasks.bar'"))
+    }
+
+    @Unroll
+    def "prints deprecation warning when removing a task with #description"() {
+        given:
+        buildFile << """
+            task foo(type: Zip) {}
+            ${code}
+        """
 
+        when:
+        executer.expectDeprecationWarning()
+        succeeds "help"
+
+        then:
+        result.assertOutputContains("The ${description} method has been deprecated. This is scheduled to be removed in Gradle 6.0. Prefer disabling the task instead, see Task.setEnabled(boolean)")
```

Comment:
this should be `assertHasCause`, according to https://builds.gradle.org/repository/download/Gradle_Check_SanityCheck/14385734:id/report-core-codenarc-integTest.html